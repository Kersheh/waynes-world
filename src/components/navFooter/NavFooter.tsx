import React from 'react';
import { useSelector, useDispatch } from 'react-redux';

import { RootState } from 'types';
import Button from 'components/button/Button';
import IconHome from 'components/icons/IconHome';
import IconHomeOutline from 'components/icons/IconHomeOutline';
import IconLibrary from 'components/icons/IconLibrary';
import IconLibraryOutline from 'components/icons/IconLibraryOutline';
import IconPlus from 'components/icons/IconPlus';
import IconPlusOutline from 'components/icons/IconPlusOutline';
import { setActiveViewAction } from 'app/appActions';
import styles from './NavFooter.module.scss';

const NavFooter = () => {
  const dispatch = useDispatch();
  const { activeView } = useSelector((state: RootState) => state.app);

  return (
    <div className={styles.navFooter}>
      <div className={styles.btns}>
        <Button
          onClick={() => dispatch(setActiveViewAction('home'))}
          styleType="fullWidth"
        >
          <div className={styles.content}>
            {activeView === 'home' ? <IconHome /> : <IconHomeOutline />}
            <div>Home</div>
          </div>
        </Button>

        <Button
          onClick={() => dispatch(setActiveViewAction('addAlbum'))}
          styleType="fullWidth"
        >
          <div className={styles.content}>
            {activeView === 'addAlbum' ? <IconPlus /> : <IconPlusOutline />}
            <div>Add Album</div>
          </div>
        </Button>

        <Button
          onClick={() => dispatch(setActiveViewAction('library'))}
          styleType="fullWidth"
        >
          <div className={styles.content}>
            {activeView === 'library' ? (
              <IconLibrary />
            ) : (
              <IconLibraryOutline />
            )}
            <div>Library</div>
          </div>
        </Button>
      </div>
    </div>
  );
};

export default NavFooter;
