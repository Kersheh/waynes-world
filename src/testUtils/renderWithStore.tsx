import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose, Reducer } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { render } from '@testing-library/react';

import rootReducer from 'store/reducers';
import rootSaga from 'store/sagas';

const renderWithStore = (component: React.ReactElement) => {
  const sagaMiddleware = createSagaMiddleware();
  const store = createStore(
    rootReducer as Reducer,
    compose(applyMiddleware(sagaMiddleware))
  );
  sagaMiddleware.run(rootSaga);

  return render(<Provider store={store}>{component}</Provider>);
};

export default renderWithStore;
