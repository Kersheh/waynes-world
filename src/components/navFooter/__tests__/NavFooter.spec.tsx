import React from 'react';

import renderWithStore from 'testUtils/renderWithStore';
import NavFooter from '../NavFooter';

describe('<NavFooter/> component', () => {
  it('should render', () => {
    const { container } = renderWithStore(
      <NavFooter />
    );

    expect(container).toMatchSnapshot();
  });
});
