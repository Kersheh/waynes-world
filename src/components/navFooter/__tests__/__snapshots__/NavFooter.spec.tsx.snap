// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavFooter/> component should render 1`] = `
<div>
  <div
    class="navFooter"
  >
    <div
      class="btns"
    >
      <button
        class="button fullWidth"
        type="button"
      >
        <div
          class="content"
        >
          <svg
            height="24"
            viewBox="0 0 512 512"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M448 463.746h-149.333v-149.333h-85.334v149.333h-149.333v-315.428l192-111.746 192 110.984v316.19z"
              fill="currentColor"
            />
          </svg>
          <div>
            Home
          </div>
        </div>
      </button>
      <button
        class="button fullWidth"
        type="button"
      >
        <div
          class="content"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 10 h-10 v-10 h-2 v10 h-10 v2 h10 v10 h2 v-10 h10 z"
              fill="white"
            />
          </svg>
          <div>
            Add Album
          </div>
        </div>
      </button>
      <button
        class="button fullWidth"
        type="button"
      >
        <div
          class="content"
        >
          <svg
            height="24"
            viewBox="0 0 512 512"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M291.301 81.778l166.349 373.587-19.301 8.635-166.349-373.587zM64 463.746v-384h21.334v384h-21.334zM192 463.746v-384h21.334v384h-21.334z"
              fill="currentColor"
            />
          </svg>
          <div>
            Library
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;
